flink-core/src/test/java/org/apache/flink/core/fs/PathTest.java

```diff
@@ -63,6 +63,15 @@ public void testPathFromString() {
 		assertEquals("/my/path", p.toUri().getPath());
 		assertEquals("file", p.toUri().getScheme());
 
+		p = new Path("C:/my/windows/path");
+		assertEquals("/C:/my/windows/path", p.toUri().getPath());
+
+		p = new Path("file:/C:/my/windows/path");
+		assertEquals("/C:/my/windows/path", p.toUri().getPath());
+
+		p = new Path("C:");
+		assertEquals("/C:", p.toUri().getPath());
```

Comment:
Well the test passes, not sure if it's semantically correct though. If it's not correct, then `Path` should be changed I guess?